core/src/main/java/com/sequenceiq/cloudbreak/service/stack/TargetedUpscaleSupportService.java

```diff
@@ -0,0 +1,58 @@
+package com.sequenceiq.cloudbreak.service.stack;
+
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.cache.annotation.Cacheable;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.altus.EntitlementService;
+import com.sequenceiq.cloudbreak.auth.crn.Crn;
+import com.sequenceiq.cloudbreak.domain.stack.Stack;
+import com.sequenceiq.cloudbreak.orchestrator.host.HostOrchestrator;
+import com.sequenceiq.cloudbreak.orchestrator.model.GatewayConfig;
+import com.sequenceiq.cloudbreak.orchestrator.model.Node;
+import com.sequenceiq.cloudbreak.service.GatewayConfigService;
+import com.sequenceiq.cloudbreak.util.StackUtil;
+
+@Service
+public class TargetedUpscaleSupportService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TargetedUpscaleSupportService.class);
+
+    @Inject
+    private GatewayConfigService gatewayConfigService;
+
+    @Inject
+    private EntitlementService entitlementService;
+
+    @Inject
+    private HostOrchestrator hostOrchestrator;
+
+    @Inject
+    private StackUtil stackUtil;
+
+    @Cacheable(cacheNames = "targetedUpscaleCache", key = "{ #stack.resourceCrn }")
+    public boolean targetedUpscaleOperationSupported(Stack stack) {
+        try {
+            // is unbound elimination supported?
+            String accountId = Crn.safeFromString(stack.getResourceCrn()).getAccountId();
+            if (!entitlementService.isUnboundEliminationSupported(accountId)) {
+                LOGGER.info("Unbound elimination is disabled for account {}, thus targeted upscale is not supported!", accountId);
+                return false;
+            }
+            // is unbound cluster config already removed?
+            GatewayConfig primaryGatewayConfig = gatewayConfigService.getPrimaryGatewayConfig(stack);
+            Set<Node> reachableNodes = stackUtil.collectReachableNodes(stack);
```

Comment:
In theory it can happen, in practice I don't think so. After unbound entitlement is on, all nodes of existing clusters will still have the conf, all nodes of new and upgraded clusters won't